src/common/src/main/scala/osmesa/common/raster/package.scala

```diff
@@ -0,0 +1,129 @@
+package osmesa.common
+import geotrellis.macros.IntTileVisitor
+import geotrellis.raster.{
+  ArrayTile,
+  CellType,
+  IntCellType,
+  IntCells,
+  IntConstantNoDataCellType,
+  IntUserDefinedNoDataCellType,
+  MutableArrayTile,
+  NoDataHandling,
+  Tile
+}
+
+import scala.collection.mutable
+
+package object raster {
+  class SparseIntTile(val cols: Int,
+                      val rows: Int,
+                      val values: Map[Long, Int],
+                      val cellType: IntCells with NoDataHandling)
+      extends ArrayTile {
+    private val noDataValue = cellType match {
```

Comment:
This is unused, as `isData()` was returning `true` for `Int.MinValue` (when `noDataValue` was used in place of `0` in `values.getOrElse(i, 0)` below).

I also suspect that I'm getting the nodata value incorrectly; I would expect to use something like `cellType.noDataValue`, but I can't find anything relevant nearby.

Reply:
This was actually correct; the problem was that `rasterize` was blindly incrementing whatever came back, even if it was nodata.